editor.planx.uk/src/pages/FlowEditor/components/Settings/Submissions/EventsLog.tsx

```diff
@@ -0,0 +1,170 @@
+        </Typography>
+        <Typography variant="body2">
+          {`If you're looking for events before January 1, 2024, please contact a PlanX developer.`}
+        </Typography>
+      </ErrorSummary>
+    );
+
+  return (
+    <TableContainer>
+      <Table>
+        <TableHead>
+          <TableRow>
+            <TableCell sx={{ width: 300 }}>
+              <strong>Event</strong>
+            </TableCell>
+            <TableCell sx={{ width: 200 }}>
+              <strong>Date</strong>
+            </TableCell>
+            <TableCell>
+              <strong>Session ID</strong>
+            </TableCell>
+            <TableCell></TableCell>
+          </TableRow>
+        </TableHead>
+        <TableBody>
+          {submissions.map((submission) => (
+            <CollapsibleRow {...submission} />
+          ))}
+        </TableBody>
+      </Table>
+    </TableContainer>
+  );
+};
+
+const CollapsibleRow: React.FC<Submission> = (submission) => {
+  const [open, setOpen] = useState<boolean>(false);
+
+  return (
+    <React.Fragment key={submission.eventId}>
+      <TableRow sx={{ "& > *": { borderBottom: "unset" } }}>
+        <TableCell
+          sx={{
+            display: "flex",
+            flexDirection: "row",
+            alignItems: "center",
+          }}
+        >
+          <Avatar
+            sx={{
+              background: "none",
+              marginRight: (theme) => theme.spacing(1),
+            }}
+          >
+            {submission.eventType === "Pay" ? (
+              <Payment
+                color={submission.status === "Success" ? "success" : "error"}
+              />
+            ) : (
+              <Send
+                color={submission.status === "Success" ? "success" : "error"}
+              />
+            )}
+          </Avatar>
+          {submission.eventType}
+        </TableCell>
+        <TableCell>
+          {format(new Date(submission.createdAt), "dd/MM/yy hh:mm:ss")}
+        </TableCell>
+        <TableCell>{submission.sessionId}</TableCell>
+        <TableCell>
+          <IconButton
+            aria-label="expand row"
+            size="small"
+            onClick={() => setOpen(!open)}
+          >
+            {open ? <KeyboardArrowUp /> : <KeyboardArrowDown />}
+          </IconButton>
+        </TableCell>
+      </TableRow>
+      <TableRow sx={{ background: (theme) => theme.palette.background.paper }}>
+        <TableCell sx={{ paddingBottom: 0, paddingTop: 0 }} colSpan={4}>
+          <Collapse in={open} timeout="auto" unmountOnExit>
+            <FormattedResponse {...submission} />
+          </Collapse>
+        </TableCell>
+      </TableRow>
+    </React.Fragment>
+  );
+};
+
+const FormattedResponse: React.FC<Submission> = (submission) => {
+  if (submission.eventType === "Pay") {
+    return (
+      <Box sx={{ margin: 1 }}>
+        <Typography variant="body2">
+          {`GOV.UK Pay reference: ${submission.eventId}`}
+        </Typography>
```

Comment:
"Pay" events currently have a null `response` in the view, but I think I can update the view to improve this a bit with a join to the `payment_status_enum` table, then we could have a more consistently formatted code block in the dropdown like : 
```
{
  status: "payment completed"
  reference: "gov-pay-id"
}
```

This should also give us that extra context of "failed how" (eg cancelled by user, rejected payment method, etc) that we currently get in Slack.

Reply:
Updated in latest db migration !